addon/record-data.js

```diff
@@ -27,18 +27,21 @@ export default class FragmentRecordData extends RecordData {
       super(identifier.type, identifier.id, identifier.clientId, store);
     }
 
+    // @patocallaghan - We need to keep a copy of the record so we can recreate the fragment/fragmentArray. There must be a better way to do this?
+    this._record = null;
     this.fragmentData = Object.create(null);
     this.serverFragments = Object.create(null);
     this.inFlightFragments = Object.create(null);
 
     this.fragments = Object.create(null);
-    this.fragmentNames = [];
+    // @patocallaghan - We need to keep the fragment definitions for later on in case we need recreate the fragment or fragmentArray
```

Comment:
`fragmentDefs` are used in https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/384/files#diff-a559085745304ba03c26083bacebf245f8ef7540dc430c48fcdb95327c49e5e1R445 when recreating the fragments/fragment arrays. There's no point keeping around `fragmentNames` as they can be inferred from `fragmentDefs`.